GHermeneus/include/GHermeneus/Machine.h

```diff
@@ -5,4 +5,185 @@
+     * @tparam SSV_T The type of the State Space Vector
+     * @tparam T  The primitive type of the State Space eq. double, int etc
+     */
+    template<typename SSV_T, typename T>
+    class Machine
+    {
+    public:
+
+        Machine() = default;
+
+        Machine(Machine<SSV_T, T>&& machine)  noexcept = default;
+
+        virtual ~Machine() = default;
+
+        /*!
+         * @brief parse the GCode to the machine instruction vector
+         * @param GCode a string_view containing the GCode
+         */
+        void parse(const std::string_view& GCode)
+        {
+            gcode = GCode;
+            extractLines(GCode); // extract the individual GCode lines
+
+            // Extract the Commands from each line this is done in an parallel unsequenced execution strategy
+            cmdlines.reserve(lines.size());
+            std::for_each(std::execution::par_unseq, lines.begin(), lines.end(), [&](Line& line) {
+                std::lock_guard<std::mutex> guard(cmdlines_mutex);
+                if (auto extractedCmd = extractCmd(line))
+                {
+                    cmdlines.emplace_back(extractedCmd.value());
+                }});
+
+            // Sort the extracted commandlines
+            std::sort(std::execution::par, cmdlines.begin(),
+                      cmdlines.end()); // Todo: check performance of diff execution policies
+        }
+
+        /*!
+         * @brief output the machines instruction vector to an output stream as in CSV format
+         * @param os Output stream
+         * @param machine A machine of type Machine<SSV_T, T>
+         * @return an output stream
+         */
+        friend std::ostream& operator<<(std::ostream& os, const Machine<SSV_T, T>& machine)
+        {
+            for (const Instruction<SSV_T, T>& cmdline : machine.cmdlines)
+            {
+                os << "line: " << cmdline.line_no << " command: " << cmdline.cmd << " Parameters ->";
+                for (const Parameter<T>& param : cmdline.params)
+                {
+                    os << " " << param.param << " = " << std::to_string(param.value);
+                }
+                os << std::endl;
+            }
+            return os;
+        };
+
+        /*!
+         * @brief Output the GCode stream to the Machines Instruction Vector
+         * @param machine The machine of type Machine<SSV_T, T>
+         * @param GCode the GCode as a string_view stream
+         * @return A machine of type Machine<SSV_T, T>
+         */
+        friend Machine<SSV_T, T>& operator<<(Machine<SSV_T, T>& machine, const std::string_view& GCode)
+        {
+            machine.parse(GCode);
+            return machine;
+        };
+
+    private:
+
+        /*!
+         * @brief extract the individual lines from the GCode
+         * @param GCode a string_view containing the GCode
+         */
+        void extractLines(const std::string_view& GCode)
+        {
+            // TODO: keep in mind CRLF and LF
```

Comment:
Maybe there's an easy way to just ignore/filter the CR part?

Reply:
I think but I haven't tried it, that `std::endl` might do the trick